### ISSUE TITLE
"Invalid state" when logging in.

### ISSUE THREAD

**Issue by KuJoe** [bangingheads/UserSpiceTwitchOAuth#1]
I've been racking my brain around this for about an hour but no luck. When I try to login the oauth_success.php exits with the "Invalid state" message. I've edited the code to display the _SESSION and _GET variables and _GET['state'] does **not** match _SESSION['oauth2state']. I checked the code against the UserSpiceDiscordOAuth plugin and they look very similar but that plugin works fine and does not produce any error.

What's strange is that when I get the "Invalid state" error, if I refresh the page it logs me in fine.

**Comment by bangingheads** [comment]
Could you note your UserSpice version/PHP version/whether you're using Apache or Nginx and I'll try to replicate the issue as it works perfectly on my configuration.

**Comment by KuJoe** [comment]
UserSpice Version = 5.1.6
PHP Version = 7.2.33 & 7.4.11 (I tried both with the same result)
Webserver Version  = Apache 2.4.46 / LiteSpeed V7.8 
OS: CloudLinux 1.1

Upon further investigation it looks like the issue is with setting the state in _SESSION so it's probably an issue on my end even though it's working fine on the Discord OAuth2 plugin. I'll report back if I can find a solution.

**Comment by KuJoe** [comment]
I figured it out:

**_SESSION['oauth2state']** is being set by both the Twitch and Discord OAuth plugins so if both are enabled whichever is loaded first will never work. I changed **oauth2state** to **twitchstate** in both twitch_oauth.php and oauth_success.php and it's working correctly now.

I compared the session data and confirmed that the bad oauth2state variable for Twitch was actually the good oauth2state variable for Discord.
